lib/record_store/provider/ns1.rb

```diff
@@ -124,10 +152,17 @@ def update(id, record, zone)
 
         # Identify the answer in this record with the matching ID, and update it
         updated = false
-        existing_record["answers"].each do |answer|
-          next if answer["id"] != id
+        existing_record["answers"].each do |existing_answer|
+          existing_answer_id = ApiAnswer.from_full_api_answer(
+            record_id: existing_record['id'],
+            type: existing_record['type'],
+            answer: existing_answer,
+          ).id
+
```

Comment:
I'll address this by making `ApiAnswer` immutable.  I don't recall changing the object's state anywhere, so this should work out.

Reply:
I misinterpretted this question. 

Here, we're just perpetuating the notion that our models are point-in-time snapshots of the entities they proxy.  For example, the existing Record types all represent resource records at their time of creation.  If the actual records change somehow, their fields won't be updated to reflect that.

In case you were wondering about this sorta ephemeral use of a record (just to get an `id`), I was originally going to map the existing answers to `ApiAnswer`s.  Then realized that I'd need to fully model answers to accomplish the goals of this method: To read answers, patch one in memory, and re-push them.  Modelling Answers enough to allow this feels like a change for another PR.

I'll just leave this as is, as `ApiAnswer`s are internally consistent.